pklloveyou/azerothcore-wotlk
# fix(CORE/Player): Equipping staff possible without enough bag space

Changes Proposed:
If the inventory is full equipping a staff/polearm is prevented if you have an offhand equipped
A proper error message "These items can be swapped" is sent to the player
The offhand no longer gets voided and sent back via mail
It seems that the problem was dead code, because the unequip was executed before the server actually checked if you can unequip the item. Unequip is handled at it's proper locations (e.g. HandleAutoEquipItemOpcode() in ItemHandler.cpp)
Issues Addressed:
Closes Equipping [fishing pole] staff possible without enough bag space #3779
SOURCE:
Description of the linked issue

Tests Performed:
Built succesful on Win10 x64
Ingame tests were succesfull
How to Test the Changes:
Create a character which can wear a staff/polearm and main+offhand
Equip a mainhand and an offhand item
Now add a staff or polearm to your inventory
Now add as many items to your inventory so that the inventory is full.
When you try to equip the staff/polearm it should be prevented
An appropriate error message gets shown
Known Issues and TODO List:
N/A

How to Test AzerothCore PRs
When a PR is ready to be tested, it will be marked as [WAITING TO BE TESTED].

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

REMEMBER: when testing a PR that changes something generic (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but especially check that the PR does not cause any regression (i.e. introducing new bugs).

For example: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but we should test Y and Z as well.



## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- 
- 


## How to Test the Changes:


1.
2.
3.

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
